lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1445,6 +1506,14 @@ def _get_csr_secret_label(self, csr: CertificateSigningRequest) -> str:
     def _get_unit_number(self) -> str:
         return self.model.unit.name.split("/")[1]
 
+    def _validate_user_provided_private_key(self, private_key: Optional[PrivateKey]) -> bool:
+        if not private_key or not private_key.is_valid():
+            logger.warning(
+                "The provided private key won't be used, a new one will be generated instead."
+            )
```

Comment:
I don't think we should log in such a helper method, the warning should be decided by whatever method calling this one.

Reply:
This function has been removed in favour of raising immediately if the key passed is not valid